src/pages/Facility/settings/devices/components/DeviceServiceHistory.tsx

```diff
@@ -0,0 +1,194 @@
+      queryParams: {
+        limit: RESULTS_PER_PAGE_LIMIT,
+        offset: ((qParams.page ?? 1) - 1) * RESULTS_PER_PAGE_LIMIT,
+      },
+      pathParams: {
+        facilityId,
+        deviceId,
+      },
+    }),
+  });
+
+  const handleEditService = (service: ServiceHistory) => {
+    setSheetState({
+      sheet: "edit",
+      serviceHistory: service,
+    });
+  };
+
+  const handleServiceCreated = () => {
+    queryClient.invalidateQueries({
+      queryKey: ["deviceServiceHistory", facilityId, deviceId, qParams],
+    });
+  };
+
+  const handleServiceUpdated = () => {
+    setSheetState({
+      sheet: "",
+      serviceHistory: null,
+    });
+
+    queryClient.invalidateQueries({
+      queryKey: ["deviceServiceHistory", facilityId, deviceId, qParams],
+    });
+  };
+
+  return (
+    <Card>
+      <CardHeader className="flex flex-row items-center justify-between">
+        <div>
+          <CardTitle>{t("service_history")}</CardTitle>
+        </div>
+        <AddServiceHistorySheet
+          facilityId={facilityId}
+          deviceId={deviceId}
+          open={isAddServiceSheetOpen}
+          setOpen={(open) =>
+            setSheetState({
+              sheet: open ? "add" : "",
+              serviceHistory: sheetState.serviceHistory,
+            })
+          }
+          onServiceCreated={handleServiceCreated}
+        />
+      </CardHeader>
+      <CardContent>
+        {isLoading ? (
+          <TableSkeleton count={5} />
+        ) : (
+          <div>
+            <Table>
+              <TableHeader>
+                <TableRow>
+                  <TableHead>{t("service_date")}</TableHead>
+                  <TableHead>{t("service_notes")}</TableHead>
+                  <TableHead className="text-right">{t("actions")}</TableHead>
+                </TableRow>
+              </TableHeader>
+              <TableBody>
+                {serviceHistory?.results?.length === 0 ? (
+                  <TableRow>
+                    <TableCell colSpan={3} className="text-center">
+                      {t("service_records_none")}
+                    </TableCell>
+                  </TableRow>
+                ) : (
+                  serviceHistory?.results.map((service: ServiceHistory) => (
+                    <TableRow key={service.id}>
+                      <TableCell className="font-medium">
+                        {format(new Date(service.serviced_on), "PPP")}
+                      </TableCell>
+                      <TableCell className="max-w-md truncate">
+                        {service.note}
+                      </TableCell>
+                      <TableCell className="text-right">
+                        <Button
+                          variant="ghost"
+                          size="icon"
+                          onClick={() => handleEditService(service)}
+                        >
+                          <Edit className="h-4 w-4" />
+                        </Button>
```

Comment:
Instead of creating complicated states, why not use `SheetTrigger` in the EditServiceHistorySheet, something like the following

```suggestion
		<EditServiceHistorySheet
		          trigger={
		          	<Button variant="ghost" size="icon">
		                          <Edit className="h-4 w-4" />
		                 </Button>
		          }
		          facilityId={facilityId}
		          deviceId={deviceId}
		          serviceRecord={service}
		          onServiceUpdated={handleServiceUpdated}
		   />
```

Reply:
consider doing the same for add sheet as well